[EDGE CIE-5316] c8yedge tool documentation -- DON'T MERGE OR DELETE (#3329)
SoftwareAG/c8y-docs: content/edge-kubernetes/quickstart.md

  - edge_server
---

> This section helps you to quickly install Cumulocity IoT Edge on a [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation) cluster using the `c8yedge` installer CLI. This tool automates the setup of K3s, Helm, Edge Operator, and the Edge instance itself.

Reviewer: ```suggestion
This section helps you to quickly install {{< product-c8y-iot >}} Edge on a [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation) cluster using the `c8yedge` installer CLI. This tool automates the setup of K3s, Helm, Edge Operator, and the Edge instance itself.
```

Author: Addressed

> ## 1. Download and install the `c8yedge` CLI

Reviewer: ```suggestion
### 1. Download and install the `c8yedge` CLI
```

Author: Addressed

Download the installer from the official source:

```shell
curl -LO https://download.cumulocity.com/edge/<c8yedge>
chmod +x c8yedge
sudo mv c8yedge /usr/bin/
```

> ## 2. View available commands

Reviewer: ```suggestion
### 2. View available commands
```

Author: Addressed

To list all available commands and options, run:

```shell
c8yedge help
```

Expected output:

```shell

> This is a CLI tool for installation and maintaining of

Reviewer: either "installing and maintaining" or "installation and maintenance"

Author: Addressed

> Cumulocity Edge on Kubernetes.

Reviewer: ```suggestion
{{< product-c8y-iot >}} Edge on Kubernetes.
```

Author: Addressed

Usage:
  c8yedge [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  help        Help about any command

>   install     Install Cumulocity Edge on Kubernetes

Reviewer: ```suggestion
  install     Install {{< product-c8y-iot >}} Edge on Kubernetes
```

Author: Addressed

>   remove      Uninstall Cumulocity Edge and Kubernetes

Reviewer: ```suggestion
  remove      Uninstall {{< product-c8y-iot >}} Edge and Kubernetes
```

Author: Addressed

>   version     Default version of Cumulocity Edge on Kubernetes to install, modify it with --version flag when calling install command

Reviewer: ```suggestion
  version     Default version of {{< product-c8y-iot >}} Edge on Kubernetes to install, modify it with --version flag when calling install command
```

Author: Addressed

Flags:
  -h, --help   help for c8yedge

Use "c8yedge [command] --help" for more information about a command.
```

> ## 3. check the version before we start (optional)

Reviewer: ```suggestion
### 3. Check the version before starting (optional)
```

Author: Addressed

> To show the version of cli, run:

Reviewer: ```suggestion
To show the version of the CLI, run:
```

```shell
c8yedge version
```

Expected output:

```shell

> Cumulocity Edge installer version: 2025.0.2

Reviewer: ```suggestion
{{< product-c8y-iot >}} Edge installer version: 2025.0.2
```

Author: Addressed

```

> ## 4. Install Edge

Reviewer: ```suggestion
### 4. Install Edge
```

Author: Addressed

Run the install command and follow the prompts. You can either pass flags explicitly or enter them interactively.

> ### Option 1: Interactive

Reviewer: ```suggestion
#### Option 1: Interactive
```

Author: Addressed

```shell
c8yedge install
```

You'll be asked to confirm system readiness. Enter `yes` to proceed.

Example output:

```shell
Set to 'yes' to confirm that system requirements are met [yes/no]: yes
Registry Host: registry.c8y.io
Registry User: <YourUsername>
Registry Password: <YourPassword>
```

> ### Option 2: With Flags

Reviewer: ```suggestion
#### Option 2: With flags
```

Author: Addressed

```shell
c8yedge install   --registry-host registry.c8y.io   --username <YourUsername>   --password  <YourPassword>   --version 2025   --confirm-system-requirements yes
```

> Once complete, you will see the operator and edge deployment running:

Reviewer: ```suggestion
Once complete, you will see the operator and Edge deployment running:
```

Author: Addressed

```shell
kubectl get pods -A
```

Expected pods:

```shell
NAMESPACE     NAME                                                   READY   STATUS    RESTARTS   AGE
c8yedge       c8yedge-operator-controller-manager-xxxx               1/1     Running   0          47s
kube-system   coredns-xxxxx                                          1/1     Running   0          49s
...
```

> ## 5. What’s next?

Reviewer: ```suggestion
### 5. What’s next?
```

Author: Addressed

Visit:

- [Verifying the Edge installation](/edge-kubernetes/installing-edge-on-k8/#verifying-the-edge-installation)
- [Accessing Edge](/edge-kubernetes/installing-edge-on-k8/#accessing-edge)

to log in and start using your Edge instance.

## 6. Uninstalling Edge
{{< c8y-admon-important >}}
Uninstalling Edge using the CLI is **non-recoverable**. It will remove both the Edge instance and the K3s cluster from the node. Backup any important data beforehand.
{{< /c8y-admon-important >}}
```shell
c8yedge remove

> Do You really want to remove Cumulocity Edge? This includes all platform data and will be non recoverable. [yes/no]: yes

Reviewer: ```suggestion
Do You really want to remove {{< product-c8y-iot >}} Edge? This includes all platform data and will be non recoverable. [yes/no]: yes
```

Author: Addressed

> To confirm Cumulocity Edge uninstallation use this random value: <random-text> , to cancel type any other string.

Reviewer: ```suggestion
To confirm {{< product-c8y-iot >}} Edge uninstallation use this random value: <random-text> , to cancel type any other string.
```

Author: Addressed

> Confirm by typing the random value from above sentence: <random-text>

Reviewer: ```suggestion
Confirm by typing the random value from the above sentence: <random-text>
```

Author: Good catch, it's a miss from our CLI itself, Making the changes here as well as in the product. Many thanks!

## Install Edge
Edge provides a command-line tool (`c8yedge`), a convenient way to install Edge on a [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation) cluster using a guided CLI-based workflow. You can download the tool from the [{{< company-c8y >}} Download Center](https://download.cumulocity.com/Cumulocity-Edge/Installer) or by running the following commands:

```shell
curl -sfL https://download.cumulocity.com/Cumulocity-Edge/Installer/c8yedge -o c8yedge
chmod +x c8yedge
sudo mv c8yedge /usr/local/bin/
```
<br/>
Before proceeding, ensure all [prerequisites](/edge-kubernetes/installing-edge-on-k8/#prerequisites) are met.

To install Edge, execute the following command and follow the interactive prompts:
```shell
sudo c8yedge install
```
{{< c8y-admon-info >}}
[Contact product support](/additional-resources/contacting-support/) to request the Edge registry credentials.
{{< /c8y-admon-info >}}

Upon successful installation, the tool will exit automatically.

> To sign in to Edge, refer to the [**Accessing Edge**](/edge-kubernetes/installing-edge-on-k8/#accessing-edge) section. The initial installation uses a bootstrap domain (`edgebootstrap.example`) along with a default license and email (`company@edgebootstrap.example`). Use this domain, email and the default credentials username “admin” and password “admin-pass” to access Edge for the first time. You can later update the bootstrap domain and license to match your environment by following the steps outlined in [**Modifying Edge**](/edge-kubernetes/manage-edge/#modify-edge).

Reviewer 2: ```suggestion
To sign in to Edge, refer to the [**Accessing Edge**](/edge-kubernetes/installing-edge-on-k8/#accessing-edge) section. The initial installation uses a bootstrap domain (`edgebootstrap.example`) along with a default license and email (`company@edgebootstrap.example`). Use this domain, email and the default credentials username “admin” and password “admin-pass” to access Edge for the first time. You can later update the domain and license to match your environment by following the steps outlined in [**Modifying Edge**](/edge-kubernetes/manage-edge/#modify-edge).
```
"bootstrap domain" isn't a C8Y term, it's only used informally in the installer.

Reviewer 2: Oh, and a similar modification to https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/files#r2114294573

Reviewer 3: Changed the text to completely remove the wording bootstrap: https://github.com/Cumulocity-IoT/c8y-docs/pull/3329/commits/b1d9c46466d3ce16d92e4a8b682ef890032e3b72

  - edge_server
---

This section helps you to quickly install {{< product-c8y-iot >}} Edge on a [Lightweight Kubernetes (K3s)](https://docs.k3s.io/installation) cluster using the `c8yedge` installer CLI. This tool automates the setup of K3s, Helm, Edge Operator, and the Edge instance itself.

### 1. Download and install Edge with `c8yedge` CLI

Download the installer from the official source:

```shell

> curl -sfLO https://download.{{< product-c8y-iot >}}.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword> --version 2025 --confirm-system-requirements yes

Reviewer 2: ```suggestion
curl -sfLO https://download.{{< product-c8y-iot >}}.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword>
```
`--version` should really be discouraged. By default, we want customers to just go to the latest fix. But we know there are exceptions to that; install the exact version in production that you tested. So we should still let them know that `--version` is there, but not in the first command we recommend.
`--confirm-system-requirements` exists as a command-line argument to help automation. I'd rather not encourage it, and instead have somebody go through the "Yes, I really did check my system configuration" gate that they're prompted for.

Reviewer 3: --username <registry username> --password <registry password>

Also add a note or INFO box below stating something like, 
"You will receive the registry credentials along with the Edge license."

Author: Totally agreed on this @rthrippleton , I will change this accordingly

Done @bbyreddy , Added the changes already as suggested.

Author: Completed

> curl -sfLO https://download.{{< product-c8y-iot >}}.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword> --version 2025 --confirm-system-requirements yes

Reviewer 2: ```suggestion
curl -sfLO https://download.cumulocity.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword> --version 2025 --confirm-system-requirements yes
```
{{< product-c8y-iot >}} is purely for branding, and may (and has in the past) change to include spaces and suchlike. So in this case, just hardcoding is safer. Looks like https://download.cumulocity.com/ appears verbatim in other parts of the Cumulocity doc, so we're in good company.

Author: Sure, I will change it right away.

Author: Completed

> curl -sfLO https://download.{{< product-c8y-iot >}}.com/Cumulocity-Edge/Installer/c8yedge && chmod +x c8yedge && ./c8yedge install --registry-host registry.c8y.io --username <YourUsername> --password <YourPassword> --version 2025 --confirm-system-requirements yes

Reviewer 2: The layout of our downloadable artifacts ought to be including the major release i.e. https://download.cumulocity.com/Cumulocity-Edge/Installer/2025/c8yedge

Because at some point in the future both 2025 and 2026 will be both supported.

Author: Right, This is still a tentative location. I will update it once we conclude the approach.

```

> Once complete, you will see the operator and Edge deployment running:

Reviewer 3: Instead of mentioning what to expect in the kubernetes (which is already explained in the "verifying-the-edge-installation" section). I suggest we show the output of the c8yedge tool, which I guess at the end shows the event "Cumulocity Edge installation is complete in 4m58s, and it's now running version 2025.0.2-xxxx"

Author: Done

```shell
kubectl get pods -A
```

Expected pods:

```shell
NAMESPACE     NAME                                                   READY   STATUS    RESTARTS   AGE
c8yedge       c8yedge-operator-controller-manager-xxxx               1/1     Running   0          47s
kube-system   coredns-xxxxx                                          1/1     Running   0          49s
...
```

### 2. What’s next?
{{< c8y-admon-info >}}
To know your c8yedge cli version, execute **`c8yedge version`**

> {{< /c8y-admon-info >}}

Reviewer 3: - [x] May be this is not required. And definitely not in the what next section.
We can add such details later when we start documenting the c8yedge tool as a standlone section in the doc.

Author: Added in the info section, if it finds itself in odd place, I am happy remove it
